StPicoDstMaker/StPicoUtilities.h

```diff
@@ -86,7 +86,78 @@ namespace StPicoUtilities {
+  // Reference multiplicity3 |eta|<1 (Pions and Kaons)
+  inline UInt_t refMult3(const UInt_t charge, const UInt_t etaId, const StMuDst& muDst)
+  {
+    // charge   0:negative, 1:positive
+    // etaId    0:East(-1<eta<0), 1:West(1>eta>0)
+
+      size_t countedTracks = 0;
+      for (Int_t itrk=0; itrk<muDst.primaryTracks()->GetEntries(); itrk++){
+      StMuTrack* track = muDst.primaryTracks(itrk) ;
+      if(!track) continue;
+
+     // these first 3 checks are easy, save time
+      const Bool_t isChargeOk = (charge==0&&track->charge()<0)||(charge==1&&track->charge()>0);
+      if (track->flag()<0 || !isChargeOk || track->nHitsFit(kTpcId)<10 ) continue; 
+
+     // check eta, a bit more elaborate
+      if (fabs(track->momentum().mag())<1.e-10) continue;
+      const Double_t eta = track->momentum().pseudoRapidity() ;
+      const Bool_t isEtaOk = (etaId==0&&(eta>-1.0&&eta<0)) || (etaId==1&&(eta>0&&eta<1.0));
+      if (!isEtaOk) continue;
+
+     // finally, check dca, nsigmaproton,mass square, if a track satisfies gets inside the if, count it
+      const Double_t p = track->momentum().mag(); 
+      const Double_t beta = track->btofPidTraits().beta();
+      float  M2;
+      if(beta==-999. ||  beta<=1.e-5 )M2=-999.;
+      else M2=p*p*(pow(1./beta, 2)-1);
+      if (track->dca().mag()<3&&track->nSigmaProton()<(-3.0) && M2<0.4) ++countedTracks;
+    }
+    return countedTracks;
+  }
+
+  //______________________________________________________________________________
+  // Reference multiplicity4 |eta|<1 (Pions and Protons)
+  inline UInt_t refMult4(const UInt_t charge, const UInt_t etaId, const StMuDst& muDst)
+  {
+    // charge   0:negative, 1:positive
+    // etaId    0:East(-1<eta<0), 1:West(1>eta>0)
+
+      size_t countedTracks = 0;
+      for (Int_t itrk=0; itrk<muDst.primaryTracks()->GetEntries(); itrk++){
+      StMuTrack* track = muDst.primaryTracks(itrk) ;
+      if(!track) continue;
+
+     // these first 3 checks are easy, save time
+      const Bool_t isChargeOk = (charge==0&&track->charge()<0)||(charge==1&&track->charge()>0);
+      if (track->flag()<0 || !isChargeOk || track->nHitsFit(kTpcId)<15 ) continue; 
+
+     // check eta, a bit more elaborate
+      if (fabs(track->momentum().mag())<1.e-10) continue;
+      const Double_t eta = track->momentum().pseudoRapidity() ;
+      const Bool_t isEtaOk = (etaId==0&&(eta>-1.0&&eta<0)) || (etaId==1&&(eta>0&&eta<1.0));
+      if (!isEtaOk) continue;
+
+     // finally, check dca, nsigmaproton,mass square, if a track satisfies gets inside the if, count it
+      const Double_t p = track->momentum().mag();
+      const Double_t beta = track->btofPidTraits().beta();
+      float  M2;
+      if(beta==-999. ||  beta<=1.e-5 )M2=-999.;
+      else M2=p*p*(pow(1./beta, 2)-1);
+      if (track->dca().mag()<3 && M2<0.4 && ((M2==-999&&(track->nSigmaKaon()>3||track->nSigmaKaon()<-3))||(M2!=-999&&(M2>0.6||M2<0.1)))) ++countedTracks;
```

Comment:
@xuji19891014, 
Do you expect these PID hardcoded numbers to change at any point? Did the bulkCorr agree that this will be the one and final definition of refMult3 and refMult4?